File: challenges/3-tipos-de-triangulo/java/diegooadb/TiposTriangulo.java
import java.util.*;

public class TiposTriangulo {
    public static void main(String[] args) {
        List<Double> doubles = new ArrayList<>(Arrays.stream(args).map(Double::parseDouble).toList());
        doubles.sort(Comparator.comparing(Double::doubleValue).reversed());

        double sideA = doubles.get(0);
        double sideB = doubles.get(1);
        double sideC = doubles.get(2);

        String format = String.format(Locale.US, "Entrada: %.1f %.1f %.1f\n", sideA, sideB, sideC);
        System.out.println(format);

        double sideBPlusSideCSquared = Math.pow(sideB, 2) + Math.pow(sideC, 2);
        double sideASquared = Math.pow(sideA, 2);

        if (sideA >= (sideB + sideC)) {
            System.out.println("Não forma quadrado");
            return;
        }

        if (sideASquared == sideBPlusSideCSquared) {
            System.out.println("TRIANGULO RETANGULO");
        }

        if (sideASquared > sideBPlusSideCSquared) {
            System.out.println("TRIANGULO OBTUSANGULO");
        }

        if (sideASquared < sideBPlusSideCSquared) {
            System.out.println("TRIANGULO ACUTANGULO");
        }

        if (sideA == sideB && sideA == sideC) {
            System.out.println("TRIANGULO EQUILATERO");
            return;
        }

        if (sideA == sideB || sideA == sideC || sideB == sideC) {
            System.out.println("TRIANGULO ISOSCELES");
        }

    }
}

File: challenges/3-tipos-de-triangulo/python/lorenzobattistela/tipostriangulo.py
SIDES = 3
sides_lst = []
for i in range(SIDES):
    sides_lst.append(float(input("Input size: ")))
sides_lst.sort(reverse=True)

sideA, sideB, sideC = sides_lst

if sideA >= (sideB + sideC):
    print('NAO FORMA TRIANGULO')
else:
    sum_of_squares = (pow(sideB, 2) + pow(sideC, 2))
    aSqrd = pow(sideA, 2)
    if aSqrd == sum_of_squares:
        print('TRIANGULO RETANGULO')
    elif aSqrd > sum_of_squares:
        print('TRIANGULO OBTUSANGULO')
    elif aSqrd < sum_of_squares:
        print('TRIANGULO ACUTANGULO')
    if sideA == sideB == sideC:
        print('TRIANGULO EQUILATERO')
    elif sideA == sideB or sideC == sideB:
        print('TRIANGULO ISOSCELES')